### PR TITLE
Write SQL migrations for new Calendly resource

### PR DESCRIPTION
**Pull request by calebschoepp**
# Changes

- New db migration to add calendly resource
- Zapatos schema updated

# Notes

I need to run these migrations on staging and prod when this merges

### CODE REVIEW

**Comment by armiantos** [comment]
From what I understand, the links are attached to interviewers and interviewees will be matched to those links right? Is there a reason why we'd go with this schema instead of 

```sql
CREATE TABLE calendly_links (
interviewer TEXT NOT NULL PRIMARY KEY REFERENCES users,
link TEXT NOT NULL
created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
);

CREATE TABLE calendly_interviews (
id UUID NOT NULL PRIMARY KEY DEFAULT uuid_generate_v4(),
interviewee TEXT REFERENCES users ON DELETE RESTRICT,
interviewer TEXT NOT NULL REFERENCES users ON DELETE RESTRICT,
);
```

I imagine we'd want to insert the volunteer link to the database when volunteers enter their calendly links, and then generate interview objects that link interviewee and interviewer when interviewees are matched.

With the current schema if we leave interviewee blank, don't we have to copy a row only to rematch the interviewer to another interviewee?

**Review by armiantos — COMMENTED** [review]
Refer to comment above

File: api/db/migrations/003_add_calendly_resource.up.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/* calendlys table */
+CREATE TABLE IF NOT EXISTS calendlys (
+    id UUID NOT NULL PRIMARY KEY DEFAULT uuid_generate_v4(),
+    link TEXT NOT NULL,
+    interviewee TEXT REFERENCES users ON DELETE RESTRICT,
```
Comment (armiantos) [inline api/db/migrations/003_add_calendly_resource.up.sql:5]: I think we want the following to support the one (interviewer) to many (interviewee) relationship
```suggestion
    interviewee TEXT[] REFERENCES users ON DELETE RESTRICT,
```

**Review by armiantos — CHANGES_REQUESTED** [review]

File: api/db/migrations/003_add_calendly_resource.up.sql (hunk 1 above)
Comment (calebschoepp) [reply api/db/migrations/003_add_calendly_resource.up.sql]: I've added this change @armiantos . Ready for rereview.

**Review by armiantos — APPROVED** [review]
LGTM! We'll lose the referential integrity but given that we'll be using the db stricly using the backend this should be enough. Great work!🚀

**Comment by calebschoepp** [comment]
> LGTM! We'll lose the referential integrity but given that we'll be using the db stricly using the backend this should be enough. Great work!

Yeah I was a bit sad about that, but this felt like the quickest approach. We can always refactor later if necessary :D.

**Pull request merged**